airflow/cli/cli_parser.py

```diff
@@ -543,6 +543,11 @@ def string_lower_type(val):
     help="Ignore depends_on_past dependencies (but respect upstream dependencies)",
     action="store_true",
 )
+ARG_WAIT_FOR_PAST_DEPENDS_BEFORE_SKIPPING = Arg(
+    ("-W", "--wait-for-past-depends-before-skipping"),
+    help="Wait for past dependencies before skipping the task when --ignore-depends-on-past is not set",
+    action="store_true",
+)
```

Comment:
I wonder if we should merge this and `--ignore-depends-on-past` into one flag, say a `--depends-on-past` option that allows three possible values `check` (default), `ignore`, and `wait`. This can be done in a subsequent, separate PR, but we should start considering the design.

Reply:
I agree. We should simplify.

Reply:
@uranusjr can you please check this [PR](https://github.com/apache/airflow/pull/28113)?
I just created a new option for Airflow CLI without changing the task instance deps classes arguments, if you prefer to merge the options everywhere, I can do that.